### PR TITLE
Make aggregation circuit generic over number of snarks

### PR DESCRIPTION
**Pull request by zhenfeizhang**
### Description

[_PR description_]

### Issue Link

[_link issue here_]

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Contents

- [_item_]

### Rationale

[_design decisions and extended information_]

### How Has This Been Tested?

[_explanation_]

<hr>

## How to fill a PR description 

Please give a concise description of your PR.

The target readers could be future developers, reviewers, and auditors. By reading your description, they should easily understand the changes proposed in this pull request.

MUST: Reference the issue to resolve

### Single responsability

Is RECOMMENDED to create single responsibility commits, but not mandatory.

Anyway, you MUST enumerate the changes in a unitary way, e.g.

```
This PR contains:
- Cleanup of xxxx, yyyy
- Changed xxxx to yyyy in order to bla bla
- Added xxxx function to ...
- Refactored ....
```

### Design choices

RECOMMENDED to:
- What types of design choices did you face?
- What decisions you have made?
- Any valuable information that could help reviewers to think critically

### CODE REVIEW

File: aggregator/src/aggregation/rlc/gates.rs [hunk 1]
```diff
@@ -130,6 +130,7 @@ impl RlcConfig {
     }
 
     #[inline]
+    #[allow(dead_code)]
```
Comment (z2trillion) [inline aggregator/src/aggregation/rlc/gates.rs:134]: I will make a separate PR that just removes all the dead code outright.

Comment (z2trillion) [reply aggregator/src/aggregation/rlc/gates.rs:134]: https://github.com/scroll-tech/zkevm-circuits/pull/1253

**Comment by lispc** [comment]
i feel it may not be necessary to use constant generic.  Anyway not bad.  Since it is already written, we can use it.  

// oh useful for testing?

File: aggregator/src/aggregation/circuit.rs [hunk 2]
```diff
@@ -448,42 +436,47 @@ impl Circuit<Fr> for AggregationCircuit {
                 || "blob checks",
                 |mut region| -> Result<(), Error> {
                     region.constrain_equal(
-                        assigned_batch_hash.num_valid_snarks.cell(),
+                        assigned_blobs.num_valid_snarks.cell(),
                         blob_data_exports.num_valid_chunks.cell(),
                     )?;
 
+                    // the following constraint is not satisfied
```
Comment (lispc) [inline aggregator/src/aggregation/circuit.rs]: what is this comment?

File: aggregator/src/aggregation/circuit.rs [hunk 3]
```diff
@@ -448,42 +436,47 @@ impl Circuit<Fr> for AggregationCircuit {
...
                         .chunk_data_digests
                         .iter()
-                        .zip_eq(assigned_batch_hash.blob.chunk_tx_data_digests.iter())
+                        .zip_eq(assigned_blobs.blob.chunk_tx_data_digests.iter())
                     {
                         for (c, ec) in chunk_data_digest
                             .iter()
                             .zip_eq(expected_chunk_data_digest.iter())
                         {
+                            println!("blob chunk tx: {:?} {:?}", c.value(), ec.value());
```
Comment (lispc) [inline aggregator/src/aggregation/circuit.rs]: we usually not use println.   log::debug? or log::trace?

File: aggregator/src/param.rs [hunk 4]
```diff
@@ -20,8 +20,8 @@ impl ConfigParams {
         Self {
             strategy: FpStrategy::Simple,
             degree: 19,
-            num_advice: vec![75],
-            num_lookup_advice: vec![10],
+            num_advice: vec![100],
+            num_lookup_advice: vec![13],
```
Comment (lispc) [inline aggregator/src/param.rs:24]: if we still use chunk num = 15, we don't need to increase nums here?

File: aggregator/src/aggregation/circuit.rs (hunk 2 above)
Comment (z2trillion) [reply aggregator/src/aggregation/circuit.rs]: it was outdated. the constraint is satisfied now.

**Comment by z2trillion** [comment]
> i feel it may not be necessary to use constant generic. Anyway not bad. Since it is already written, we can use it.
> 
> // oh useful for testing?

Yes, it is mainly for testing. The other option I considered was to use feature flags to vary MAX_AGG_SNARKS, but I thought that that would be harder to maintain.

**Comment by lispc** [comment]
Seems you have to increase columns a bit. Try increase as small as possible

File: aggregator/src/param.rs (hunk 4 above)
Comment (z2trillion) [reply aggregator/src/param.rs:24]: I am looking into what changes caused the increased column usage. The params were set to their previous values here: increased https://github.com/scroll-tech/zkevm-circuits/pull/1225/commits/edb0ae70e6dae4914d80387da51494effcf195d7

**Comment by z2trillion** [comment]
> Seems you have to increase columns a bit. Try increase as small as possible

The existing change by zhenfei is the minimum necessary.

**Review by lispc — APPROVED** [review]

**Pull request merged**